docs/database/database-metrics.md

```diff
@@ -436,6 +571,432 @@ This metric is [recommended][MetricRecommended].
+
+Refer `db.cosmosdb.operation.request_charge` metrics for dimensions.
+
+### Metric: `db.query.limit`
+
+It captures the requested page size for Query or Feed operations in Azure Cosmos DB. It helps in identifying patterns in page size requests that could contribute to higher latencies or other performance issues.
+
+This metric SHOULD be specified with
+[`ExplicitBucketBoundaries`](https://github.com/open-telemetry/opentelemetry-specification/tree/v1.35.0/specification/metrics/api.md#instrument-advisory-parameters)
+of `[50, 100, 250, 500, 1000, 2000, 5000]`.
+
+Explaining bucket configuration:
+
+1. 50, 100, 250, 500: These small page sizes are common for lightweight queries where quick responses are essential, such as in real-time applications.
+2. 1000, 2000: These boundaries capture scenarios where larger datasets are returned in fewer queries, which can be more efficient but might result in longer query execution times.
+3. 5000: This boundary is used to capture rare cases where very large page sizes are requested. Although Cosmos DB generally handles larger datasets efficiently, large page sizes can sometimes lead to performance bottlenecks due to increased processing times or network bandwidth limitations.
+
+Refer `db.cosmosdb.operation.request_charge` metrics for dimensions.
+
+### Metric: `db.query.response.item_count`
+
+It captures the number of items returned by a query or feed operation in Azure Cosmos DB. It helps identify response sizes that may contribute to high latency, increased memory/CPU usage, or network call failures.
+
+This metric SHOULD be specified with
+[`ExplicitBucketBoundaries`](https://github.com/open-telemetry/opentelemetry-specification/tree/v1.35.0/specification/metrics/api.md#instrument-advisory-parameters)
+of `[10, 50, 100, 250, 500, 1000, 2000, 5000, 10000]`.
+
+Explaining bucket configuration:
+
+1. 10, 50, 100: These buckets are useful for capturing scenarios where only a small number of items are returned. Such small queries are common in real-time or interactive applications where performance and quick responses are critical.
+2. 250, 500, 1000: These values represent typical workloads where moderate amounts of data are returned in each query.
+3. 2000, 5000: These boundaries capture scenarios where the query returns large datasets. These larger page sizes can potentially increase memory or CPU usage and may lead to longer query execution times, making it important to track performance in these ranges.
+4. 10000: This boundary is included to capture rare, very large response sizes. Such queries can put significant strain on system resources, including memory, CPU, and network bandwidth, and can often lead to performance issues such as high latency or even network drops.
+
+Refer `db.cosmosdb.operation.request_charge` metrics for dimensions.
```

Comment:
this does not look right in generic metric

Reply:
As you suggested, moved this to `cosmosdb.md`

Reply:
it somehow got lost